providers/yaml/src/main/java/org/jboss/resteasy/plugins/providers/YamlProvider.java

```diff
@@ -110,4 +143,131 @@ public void writeTo(Object t, Class<?> type, Type genericType, Annotation[] anno
 
    }
 
+   private static ClassLoader getClassLoader(final Class<?> type) {
+      if (System.getSecurityManager() == null) {
+         // Get the TCCL first
+         ClassLoader result = Thread.currentThread().getContextClassLoader();
+         if (result == null) {
+            result = type.getClassLoader();
+         }
+         return result;
+      }
+      return AccessController.doPrivileged((PrivilegedAction<ClassLoader>) () -> {
+         // Get the TCCL first
+         ClassLoader result = Thread.currentThread().getContextClassLoader();
+         if (result == null) {
+            result = type.getClassLoader();
+         }
+         return result;
+      });
+   }
+
+   private static Collection<String> createAllowedList() {
+      final String value = getProperty(ALLOWED_LIST);
+      return value == null ? Collections.emptyList() : Arrays.asList(value.split(","));
+   }
+
+   private static boolean isTypeCheckDisabled() {
+      final String value = getProperty(DISABLE_TYPE_CHECK);
+      return value != null && (value.isEmpty() || value.equalsIgnoreCase("true"));

```

Comment:
This accepts an empty value `""` as true.  Should it only accept not empty "true" value as true ?

Reply:
We could require it be set to `true` explicitly. However, this allows something like `-Dresteasy.yaml.deserialization.disable.type.check` to be passed.